kmir/src/kmir/kdist/mir-semantics/symbolic/spl-token.md

```diff
@@ -0,0 +1,460 @@
+           )
+        => #traverseProjection(
+             DEST,
+             VALUE,
+             PROJS,
+             CtxSPLDataBuffer CTXTS
+           )
+        ...
+       </k>
+    requires PROJS =/=K .ProjectionElems
+    [priority(30)]
+
+  rule <k> #traverseProjection(
+             DEST,
+             SPLDataBorrow(PLACE, SPLDataBuffer(VALUE)),
+             PROJS,
+             CTXTS
+           )
+        => #traverseProjection(
+             DEST,
+             VALUE,
+             PROJS,
+             CtxSPLDataBorrow(PLACE) CTXTS
+           )
+        ...
+       </k>
+    requires PROJS =/=K .ProjectionElems
+    [priority(30)]
+
+  rule <k> #traverseProjection(
+             DEST,
+             SPLDataBorrowMut(PLACE, SPLDataBuffer(VALUE)),
+             PROJS,
+             CTXTS
+           )
+        => #traverseProjection(
+             DEST,
+             VALUE,
+             PROJS,
+             CtxSPLDataBorrowMut(PLACE) CTXTS
+           )
+        ...
+       </k>
+    requires PROJS =/=K .ProjectionElems
+    [priority(30)]
+
+  rule #buildUpdate(VAL, CtxSPLDataBuffer CTXS)
+    => #buildUpdate(SPLDataBuffer(VAL), CTXS)
+  rule #buildUpdate(VAL, CtxSPLDataBorrow(PLACE) CTXS)
+    => #buildUpdate(SPLDataBorrow(PLACE, SPLDataBuffer(VAL)), CTXS)
+  rule #buildUpdate(VAL, CtxSPLDataBorrowMut(PLACE) CTXS)
+    => #buildUpdate(SPLDataBorrowMut(PLACE, SPLDataBuffer(VAL)), CTXS)
+
+  rule #projectionsFor(CtxSPLDataBuffer CTXS, PROJS) => #projectionsFor(CTXS, PROJS)
+  rule #projectionsFor(CtxSPLDataBorrow(_) CTXS, PROJS) => #projectionsFor(CTXS, PROJS)
+  rule #projectionsFor(CtxSPLDataBorrowMut(_) CTXS, PROJS) => #projectionsFor(CTXS, PROJS)
+```
+
+## RefCell::<&mut [u8]> helpers
+
+```k
+  rule [spl-borrow-data]:
+    <k> #execTerminator(terminator(terminatorKindCall(FUNC, operandCopy(place(LOCAL, PROJS)) .Operands, DEST, someBasicBlockIdx(TARGET), _UNWIND), _SPAN))
+      => #mkSPLBorrowData(
+           DEST,
+           operandCopy(place(LOCAL, PROJS)),
+           place(LOCAL, PROJS),
+           #isSPLBorrowMutFunc(#functionName(lookupFunction(#tyOfCall(FUNC))))
+         )
+         ~> #execBlockIdx(TARGET)
+    ...
+    </k>
+    requires #isSPLBorrowFunc(#functionName(lookupFunction(#tyOfCall(FUNC))))
+    [priority(30), preserves-definedness]
+
+  syntax KItem ::= #mkSPLBorrowData ( Place , Evaluation , Place , Bool ) [seqstrict(2)]
+
+  rule <k> #mkSPLBorrowData(DEST, SPLRefCell(PLACE, BUF), _SRC, false)
+        => #setLocalValue(DEST, SPLDataBorrow(PLACE, BUF))
+        ...
+       </k>
+
+  rule <k> #mkSPLBorrowData(DEST, SPLRefCell(PLACE, BUF), _SRC, true)
+        => #setLocalValue(DEST, SPLDataBorrowMut(PLACE, BUF))
+        ...
+       </k>
+```
+
+## Vec::<u8> helpers
+
+```k
+  rule [spl-vec-as-slice-copy]:
+    <k> #execTerminator(terminator(terminatorKindCall(FUNC, OP:Operand _OPS:Operands, DEST, someBasicBlockIdx(TARGET), _UNWIND), _SPAN))
+      => #setLocalValue(DEST, OP) ~> #execBlockIdx(TARGET)
```

Comment:
Similar situation here, `_OPS` is unused and should always just be `.Operands` if the function is `_.as_slice()`